### PR TITLE
Fix invalid state, which happen if signals raise exception

### PR DESCRIPTION
**Pull request by Andertaker**

### CODE REVIEW

**Comment by vdboor** [comment]
I wonder, won't it work if the whole `.save()` method was wrapped inside a transation?

File: threadedcomments/models.py [hunk 1]
```diff
@@ -27,21 +31,35 @@ def root_id(self):
...
-        super(ThreadedComment, self).save(*args, **kwargs)
         if skip_tree_path:
+            super(ThreadedComment, self).save(*args, **kwargs)
             return None
 
+        with transaction.atomic():
+            if self.submit_date is None: # for comment save
+                self.submit_date = timezone.now()
+            Comment.objects.bulk_create([self])
+            c = Comment.objects.latest("id")
```
Comment (HonzaKral) [inline threadedcomments/models.py]: this is by no means safe, it is a race condition

**Comment by HonzaKral** [comment]
This is not a valid fix, it introduces several other issues, not to mention raw SQL. Proper handling of this is to encourage use of transaction middleware.

**Comment by vdboor** [comment]
Indeed. This needs to be solved in a better way.

**Comment by Sonicmr** [comment]
I am having the same problem here, anyone fixed it yet ?

**Comment by Sonicmr** [comment]
lol someone replied and deleted his comment the next day

**Comment by Sonicmr** [comment]
any help ?

**Comment by Sonicmr** [comment]
This fix is breaking ajax when replying

**Pull request closed without merging**